src/spartan/ppsnark.rs

```diff
@@ -47,36 +47,36 @@ fn padded<E: Engine>(v: &[E::Scalar], n: usize, e: &E::Scalar) -> Vec<E::Scalar>
 #[derive(Clone, Serialize, Deserialize)]
 #[serde(bound = "")]
 pub struct R1CSShapeSparkRepr<E: Engine> {
-  N: usize, // size of the vectors
+  pub(in crate::spartan) N: usize, // size of the vectors
 
   // dense representation
-  row: Vec<E::Scalar>,
-  col: Vec<E::Scalar>,
-  val_A: Vec<E::Scalar>,
-  val_B: Vec<E::Scalar>,
-  val_C: Vec<E::Scalar>,
+  pub(in crate::spartan) row: Vec<E::Scalar>,
```

Comment:
Why do these fields need to be made visible within spartan module?

Reply:
The `batched_ppsnark`, which is the batched variant of this file, needs access to the fields in order to construct sumcheck instances and evaluations (similarly to what ppsnark itself does).